src/core/lombok/javac/handlers/JavacHandlerUtil.java

```diff
@@ -1627,7 +1636,8 @@ public static boolean hasNonNullAnnotations(JavacNode node) {
 	public static boolean hasNonNullAnnotations(JavacNode node, List<JCAnnotation> anns) {
 		if (anns == null) return false;
 		for (JCAnnotation ann : anns) {
-			for (String nn : NONNULL_ANNOTATIONS) if (typeMatches(nn, node, ann)) return true;
```

Comment:
It seemed wrong that only `ann` was used here instead of `ann.annotationType`. Fixed this alongside.

Reply:
The difference would be `@NonNull` with `ann` vs. `NonNull` with `ann.annotationType`.

Reply:
I've tried to find a way to have this use `ann` vs `ann.annotationType` difference actually produce different results, but I haven't managed. Let's roll with this update, it makes more sense, it _is_ about the type after all.

Reply:
Note: this change was a breaking change for some specific use case. More info in https://github.com/projectlombok/lombok/issues/3717